tools/config/go-templates/client.mustache

```diff
@@ -141,14 +126,6 @@ func parameterAddToHeaderOrQuery(headerOrQueryParams any, keyPrefix string, obj
 				value = "invalid"
 
 			case reflect.Struct:
-				if t,ok := obj.(modelWithMap); ok {
```

Comment:
toMap is not being implemented

Reply:
None of the models implement it so this code does not execute. 
If we had any objects as parameters (currently that is not possible to express in swagger core) we would need to add JSON marshaler here. 

Separately. We initially had approach that minimized changes in the templates and flagged specific areas we change: https://github.com/mongodb/atlas-sdk-go/blob/main/tools/CONTRIBUTING.md#modification-of-the-custom-templates

This approach was to be able to continue update SDK with upstream. 
If your preference is to finetune templates we should likely remove {{! X-XGEN-MODIFIED }} and look to remove any dead code from templates (there is much more that we left in order to not make any unnecessary changes in templates.

Reply:
SGTM